nemo/collections/tts/modules/waveglow.py

```diff
@@ -101,7 +110,7 @@ def forward(self, spec, audio=None, run_inverse=True, sigma=1.0):
         if run_inverse:
             # norm_dist_to_audio is used to predict audio from spectrogram so only used in val or infer mode
             # Could also log train audio but currently not done
-            audio_pred = self.norm_dist_to_audio(spec=spec, sigma=sigma)
+            audio_pred = self.norm_dist_to_audio(spec=spec, sigma=sigma, z=z)
```

Comment:
Why add z here? This breaks waveglow evaluation since this is no longer the z that you pass to `forward`. It is the `z` that gets computed during the forward pass.

Reply:
I have added optional z parameter to forward() as Jarvis needs it. It has to be None in this case - but clearly the error happens in 'not None' conditional section ... This may be PT bug : I am trying to rename outer scope 'z' variable first - if that does not fix things, I will change to separate calls for inference and validation mode.

Reply:
Oh, figured - no PT bug mystery here - there was 'z' local variable assigned right there in forward - but only for validation. Renamed local variable, should be all good now.